extra/mariabackup/backup_copy.cc

```diff
@@ -559,9 +559,9 @@ datafile_read(datafile_cur_t *cursor)
 		return(XB_FIL_CUR_EOF);
 	}
 
-	if (!os_file_read(IORequestRead,
+	if (os_file_read(IORequestRead,
 			  cursor->file, cursor->buf, cursor->buf_offset,
-			  to_read)) {
+			  to_read) != DB_SUCCESS) {
```

Comment:
The incorrect code in `mariabackup` must be fixed in 10.2 already.

Please extend MDEV-19231 with a report that we are ignoring failures in `os_file_read()` in a few cases. Try to cover these cases with tests. See my other comment.

Reply:
Success path of this is covered by tests. So, when return value is not `DB_SUCCESS` a lot of tests failed.